### PR TITLE
Fix: Certificates in general

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

I noticed multiple problems with certificates on custom domains, and addressed them all in this PR.

#### ✅ 1. Allow using same domain on multiple projects

Currently, your second project would be stuck at `in progress` of cert creation, because `Renew isn't required`, which is true. This PR makes sure to set certificate ID on the domain, even tho renew isn't required.

https://user-images.githubusercontent.com/19310830/159718472-8c29373a-d6bb-4a90-a45e-fbd276f5b7bc.mp4

#### ✅ 2. Delete certificate document when deleting certificate file

We currently only remove files, which results in Appwrite still thinking certificate is there (and renewal isn't required), even tho file is already missing and Traefik cannot communicate over HTTPS.

https://user-images.githubusercontent.com/19310830/159723802-5ae4466d-8302-404a-beb4-4c5405a24b34.mp4

#### ✅ 3. Only delete certificate if no other domain is using it

With step 2 fixed, now if you delete a domain, it deletes the certificate too. Problem is, another project could still be using it. Let's not remove it! In deletes worker, we check if no domain is using certificate before deleting it.

https://user-images.githubusercontent.com/19310830/159730642-4f9560d4-6239-4f88-a048-ef218964a925.mp4

## Test Plan

The PR was created from DO droplet, where I was manually testing the solution is working before pushing to git.

## Related PRs and Issues

https://github.com/appwrite/appwrite/issues/2933

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

✅

### CODE REVIEW

File: app/workers/deletes.php [hunk 1]
```diff
@@ -113,6 +113,8 @@ public function run(): void
                 Console::error('No delete operation for type: ' . $type);
                 break;
         }
+
+        Authorization::reset();
```
Comment (eldadfux) [inline app/workers/deletes.php:117]: Why did we change the Authorization behavior? usually in worker context we want to act as admin and have no permission check.

Comment (Meldiron) [reply app/workers/deletes.php:117]: Initially, I wanted to only disable it at the top of the file, but that didn't work for some reason - some other files must have reseted it somewhere.

What I did is I copied the solution from another worker (certificates), to disable at the beginning of the run, and reset at the end.

Comment (TorstenDittmann) [reply app/workers/deletes.php:117]: Maybe there is some calls in the database library itself?

Comment (TorstenDittmann) [reply app/workers/deletes.php:117]: Maybe the db library itself has calls to reset it?

Comment (TorstenDittmann) [reply app/workers/deletes.php:117]: Maybe the db library itself has calls to reset it?

**Comment by arbile26** [comment]
I'm facing the same problem....

File: app/workers/deletes.php (hunk 1 above)
Comment (Meldiron) [reply app/workers/deletes.php]: A problem was caused when library called reset, yes. It is now fixed, please revisit.

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**